wagtail/admin/views/pages/create.py

```diff
@@ -347,16 +359,13 @@ def get_context_data(self, **kwargs):
                 "parent_page": self.parent_page,
                 "edit_handler": bound_panel,
                 "action_menu": action_menu,
-                "side_panels": side_panels,
+                "side_panels": self.get_side_panels(),
                 "form": self.form,
                 "next": self.next_url,
                 "has_unsaved_changes": self.has_unsaved_changes,
                 "locale": None,
                 "translations": [],
-                "media": bound_panel.media
-                + self.form.media
-                + action_menu.media
-                + side_panels.media,
+                "media": bound_panel.media + self.form.media + action_menu.media,
```

Comment:
I'm slightly uneasy about us not adding media from the side panels here... while none of the panels use media right now (as far as I can see), it's part of the contract of using components, and if this changes in future (e.g. we add a hook for inserting custom panels) we're liable to end up with partially broken functionality. I don't think this will add _too_ much clutter to this method...

    side_panels = self.get_side_panels()
    media = bound_panel.media + self.form.media + action_menu.media
    for panel in side_panels:
        media += panel.media

Reply:
Yep, good shout. Added in b363989. I introduced a `MediaContainer` class in 46c524a to allow combining the media definitions in a more declarative way. I can see us using this in other places as well. Not sure if that's OK (suggestions for naming and where to put it are welcome), if not I can revert it and use the for loop with `+` everywhere.

As you said, the side panels currently don't have any media definitions so I don't see a way to write a test for this. We can also write unit test for the `MediaContainer` class, but not sure if that can be done without digging too much into Django's `Media` class, and our other unit tests probably have already covered the real-life use case by ensuring the media files for the other objects are rendered...